rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,417 @@
+            event.counterparty = CPT_RAINBOW_SWAPS
+            event.address = self.router_address
+
+    def _find_event_by_log(
+            self,
+            logs: list['EvmTxReceiptLog'],
+            event_signature: bytes,
+    ) -> EvmTxReceiptLog | None:
+        """
+        Find the log by the event signature.
+
+        Args:
+            logs (list[EvmTxReceiptLog]): List of logs.
+            event_signature (bytes): Event signature to be searched.
+
+        Returns:
+            Optional[EvmTxReceiptLog]: Log if found, else None.
+        """
+        for tx_log in logs:
+            if tx_log.topics[0] == event_signature:
+                return tx_log
+        return None
+
+    def _get_in_out_events(
+            self,
+            events: list['EvmEvent'],
+            sender: ChecksumEvmAddress,
+    ) -> tuple[Optional['EvmEvent'], Optional['EvmEvent']]:
+        """
+        Gets the in and out event in the list of events
+        Args:
+            events (list['EvmEvent']): A list of EVM events to be processed.
+            sender (ChecksumEvmAddress): The address of the sender involved in the events.
+        Returns:
+            in_event: Represents the receive the tokens from the swap
+            ou_event: Represents the spend the tokens to swap
+        """
+        in_event = out_event = None
+        for event in events:
+            if (
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == self.evm_inquirer.native_token
+            ):
+                # The "send eth" event is not enriched by enricher_rules,
+                # so it needs to be enriched here
+                self._enricher_event(event, sender)
+
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.RECEIVE and
+                event.counterparty == CPT_RAINBOW_SWAPS
+            ):
+                in_event = event
+            elif (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND and
+                event.counterparty == CPT_RAINBOW_SWAPS
+            ):
+                out_event = event
+
+        return in_event, out_event
+
+    def _create_fee_event(
+            self,
+            tx_hash: EVMTxHash,
+            timestamp: Timestamp,
+            fee_raw: int,
+            fee_asset: CryptoAsset,
+            sender: ChecksumEvmAddress,
+    ) -> 'EvmEvent':
+        """Create a new fee event for rainbow swap
+
+        Args:
+            tx_hash (EVMTxHash): transaction hash
+            timestamp (int): transaction timestamp
+            fee_raw (int): amount of fee charged in wei
+            fee_asset (Asset): fee asset
+            sender (ChecksumEvmAddress): sender address
+        """
```

Comment:
the arguments are explicit I believe you can simplify this docstring

```suggestion
        """Create a new fee event for a rainbow swap"""
```